pkg/controller/cluster/main-controller.go

```diff
@@ -767,6 +784,13 @@ func (c *Controller) syncHandler(key string) error {
 					return err
 				}
 			}
+
+			// Set the PVC purge  behavior needed.
+			err = c.purgePVCsOnDelete(ctx, mi)
```

Comment:
why do we need this here if we are doing it as well on `processDelete`

Reply:
It should be harmless to do this here and keeps the PVC reflecting the owernship even if it is not deleted.